File: Modules/Common/AbnormalInfo/SIE.Web.AbnormalInfo/AbnormalMonitors/AbnormalMonitorTasks/Commands/AddTaskCommand.js
SIE.defineCommand('SIE.Web.AbnormalInfo.AbnormalMonitors.Commands.AddTaskCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit", iconCls: "iconfont icon-AddEntity icon-green" },

    execute: function (view, source) {

        var me = this;
        var indata = {};
        var entity = me.getEditEntity();
        indata.Data = Ext.encode(entity.data);
        view.execute({
            data: indata,
            success: function (res) {
                CRT.Workbench.addPage({
                    entityType: me.view.model,
                    recordId: entity.data.Id,
                    title: me.getEditViewTitle(entity),
                    isDetail: true,
                    params: {
                        Code: res.Result.Code,
                        TaskType: res.Result.TaskType,
                        TaskState: res.Result.TaskState,
                        IsAdd: true,
                    }
                });
            }
        });
    },
});

File: Modules/Common/AbnormalInfo/SIE.Web.AbnormalInfo/AbnormalMonitors/AbnormalMonitorTasks/Commands/AddTaskCommand.cs
using SIE.AbnormalInfo.AbnormalMonitors;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Text;

namespace SIE.Web.AbnormalInfo.AbnormalMonitors.Commands
{
    /// <summary>
    /// 添加异常任务
    /// </summary>
  public  class AddTaskCommand : ViewCommand<ViewArgs>
    {
        /// <summary>
        /// 执行方法
        /// </summary>
        /// <param name="args">视图参数</param>
        /// <param name="scope">SIE.MetaModel.View.Block.EntityType</param>
        /// <returns>object</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var bill = args.Data.ToJsonObject<AbnormalMonitorTask>();
            if (null == bill)
                throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(bill)));
            bill.PersistenceStatus = Domain.PersistenceStatus.New;
            bill.Code = RT.Service.Resolve<AbnormalMonitorTaskService>().GenerateCode();
            bill.TaskType = SIE.AbnormalInfo.Common.TaskType.Manual;
            bill.TaskState = SIE.AbnormalInfo.Common.TaskStateEnum.ToDo;
            return bill;
        }
    }
}
